src/python/pants/engine/internals/parametrize_test.py

```diff
@@ -73,6 +86,41 @@ def test_to_parameters_failure(exception_str: str, args: list[Any], kwargs: dict
             ],
             {"f": Parametrize("b", "c"), "x": Parametrize("d", "e")},
         ),
+        (
+            [
+                ("a@parametrize=A", {"f0": "c", "f1": 1, "f2": 2}),
+                ("a@parametrize=B", {"f0": "c", "f1": 3, "f2": 4}),
+            ],
+            {
+                "f0": "c",
+                **Parametrize("A", f1=1, f2=2),
+                **Parametrize("B", f1=3, f2=4),
+            },
+        ),
+        (
+            [
+                ("a@parametrize=A", {"f": 1}),
+                ("a@parametrize=B", {"f": 2}),
+                ("a@parametrize=C", {"f": "x", "g": ()}),
+            ],
+            {
+                # Field overridden by some parametrize groups.
+                "f": "x",
+                **Parametrize("A", f=1),
+                **Parametrize("B", f=2),
+                **Parametrize("C", g=[]),
+            },
+        ),
+        (
+            [
+                ("a@parametrize=A", {"f": "x", "a": 2, "c": 4}),
+            ],
+            {
+                "f": "x",
+                **Parametrize("A", a=1, b=3),
+                **Parametrize("A", a=2, c=4),
+            },
```

Comment:
this is constructing a dict which explicitly allows unpacked keys to replace previous keys. I suspect Python will report an error when doing:

```python
            target(
                f="x",
                **Parametrize("A", a=1, b=3),
                **Parametrize("A", a=2, c=4),
            )
```

Reply:
```python
>>> def f(**kwargs):
...     return
... 
>>> f(**{"a": 1}, **{"a": 2})
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
TypeError: f() got multiple values for keyword argument 'a'
```

Reply:
🤦🏽 thanks Jacob :)